electroncash_gui/qt/token_send.py

```diff
@@ -492,45 +615,154 @@ def on_reset(b, commitment_hex=commitment_hex, le=le):
+                multiplier_sb.setMaximum(10_000)
+                multiplier_sb.setSuffix(" " + _("copies"))
+                multiplier_sb.setSpecialValueText(_("Single"))
+                multiplier_sb.setValue(copies)
+
+                def on_multiplier_change(value, _row_num=row_num):
+                    self.nfts_to_mint[_row_num]["copies"] = value
+                multiplier_sb.valueChanged.connect(on_multiplier_change)
+                hbox.addWidget(multiplier_sb)
+                tw.setItemWidget(item, self.ColsMint.multiplier, w)
+
+                row_num += 1
+        else:
+            for tid, names in self.token_nfts.items():
+                if not names:
+                    # Defensive programming: should never happen
+                    continue
+                if len(names) == 1:
+                    # This group has only 1 item, push to top-level and don't build a sub-item
+                    item = add_leaf_item(tw.invisibleRootItem(), tid, names[0])
+                    # Subscribe to counts label updates
+                    item.setData(0, self.DataRoles.receives_nft_count_updates, True)
+                    continue
+                # This group has more than 1 item, build a subgrouping
+                parent = QtWidgets.QTreeWidgetItem(["", tid, "", ""])
+                parent.setIcon(self.ColsNFT.token_id, self.token_meta.get_icon(tid))
+                parent.setToolTip(self.ColsNFT.token_id, tid)
+                parent.setFlags((parent.flags() | QtCore.Qt.ItemIsAutoTristate | QtCore.Qt.ItemIsUserCheckable)
+                                & ~QtCore.Qt.ItemIsSelectable)
+                parent.setData(0, self.DataRoles.token_id, tid)
+                parent.setData(0, self.DataRoles.receives_nft_count_updates, True)  # Subscribe to update of counts label
+                parent.setData(0, self.DataRoles.receives_nft_flag_updates, True)  # Subscribe to updates of Flags label
+                parent.setCheckState(0, QtCore.Qt.Checked)  # Needs to be called once to make checkbox appear
+                nfts_selected = self.token_nfts_selected.get(tid, set())
+                nchkd = len(nfts_selected)
+                ntot = len(names)
+                if nchkd == 0:
+                    parent.setCheckState(0, QtCore.Qt.Unchecked)
+                elif nchkd < ntot:
+                    parent.setCheckState(0, QtCore.Qt.PartiallyChecked)
+                tw.addTopLevelItem(parent)
+                for name in names:
+                    add_leaf_item(parent, tid, name)
+
+            if tw.topLevelItemCount() == 1:
+                # Auto-expand if only 1 item
+                item = tw.topLevelItem(0)
+                if item.childCount() > 0:
+                    item.setExpanded(True)
+
+        # In send and edit mode, if we have no NFTs, hide this widget completely
+        if self.form_mode in (self.FormMode.send, self.FormMode.edit):
+            self.gb_nft.setHidden(not self.have_nfts())
```

Comment:
How can one be in edit mode and not have nfts?

Reply:
Ah, this was me semi-automatically adjusting things when I added the 3rd "mint" form mode.  And since this hide logic would run in both modes initially, I made the condition match that and updated the comment.  Totally agree it's misleading now though so I'll adjust!

Reply:
Yeah I actually got the refactor mental pattern you used like .. "everything not mint" -> inverted.. I imagined that's what you did, heh.